Dockerfile

```diff
@@ -13,33 +14,34 @@ RUN set -eux ; \
     rm -rf /var/lib/apt/lists/*
 
 # install pipenv for dependency management
-# TODO fix the base python image so that it creates a home directory for the vdnonroot user
-# USER vdbnonroot
-ENV LANG="en_US.UTF-8"
-ENV LC_ALL="en_US.UTF-8"
 RUN pip install pipenv
 
 # Switch to mpyl source code directory
 WORKDIR /app/mpyl
 
 # Install the project dependencies.
+USER vdbnonroot
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
 COPY Pipfile Pipfile.lock ./
-RUN pipenv sync --system
+RUN pipenv sync
 
 # Copy the source code into the container.
-COPY --link src/mpyl ./
-
-# Set pythonpath for mpyl
-ENV PYTHONPATH=/app
-
-# Switch to the directory of the calling repo
-WORKDIR /repo
+COPY --link --parents src/mpyl ./
 
+# Make sure Python doesn't expect src. in the package names
+ENV PYTHONPATH=/app/mpyl/src
+# Use MPyL's Pipfile, not the one from the caller repo
+ENV PIPENV_PIPFILE=/app/mpyl/Pipfile
 # Prevents Python from writing pyc files.
 ENV PYTHONDONTWRITEBYTECODE=1
 # Keeps Python from buffering stdout and stderr to avoid situations where
 # the application crashes without emitting any logs due to buffering.
 ENV PYTHONUNBUFFERED=1
 
+## Switch to the directory of the caller repo (must be mounted while running)
+WORKDIR /repo
+
 # Run the application.
-ENTRYPOINT ["python", "/app/mpyl/__main__.py"]
+ENTRYPOINT ["pipenv", "run", "python", "/app/mpyl/src/mpyl/__main__.py"]
```

Comment:
are these changes related to the non root user?

Reply:
Yes. Before we would [install `pipenv` dependencies to the entire system](https://github.com/Vandebron/gh-mpyl/pull/112/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557L27), now we install it only for the non-root user. This requires us to trigger the application through pipenv (like we do locally).